loader/loader_test.go

```diff
@@ -857,6 +857,10 @@ func TestSplitPrefix(t *testing.T) {
 			wantParts: []string{"foo"},
 			wantPath:  "/bar",
 		},
+		{
+			input:    "c:/a/b/c",
+			wantPath: "c:/a/b/c",
```

Comment:
The prefix of a path will affect the ref/path under which the file's data will be found under the root `data` document during eval-time. E.g. if you load json data with `c:/my_data.json`, then any data loaded from the file found at `/my_data.json` will be found under the path `data.c` when referenced in a Rego file. Unfortunately, a path prefix can consist of a single letter, and as this added test shows, this PR constitutes a breaking change in expected behavior, as we'd expect the prefix `c` and the path `/a/b/c`.

Reply:
@johanfylling This is the weird world of Windows paths, unfortunately. 🤭

I've encountered this issue while trying to load TLS certificates for the OPA server.
I did not know about the `file:///c:/a/b/c` that should be used, and made the assumption that the drive letter is just ignored. Not sure if it's specified in the documentation that this specific file format should be used for Windows absolute paths.

However, this doesn't fix the TLS certificate load problem.
If I only load the bundle using the absolute file format (`file:///c:/a/b/c`) I have no issues. 
However, if I use the same absolute path for loading the TLS certificates, then I get the following error 
`error: open file:///C:/Temp/certs/certificate.pem: The filename, directory name, or volume label syntax is incorrect.`

I think the problem might be somewhere else, what do you think?

Reply:
The documentation for this might need some clarification; you're not the first to stumble onto this quirk. We'll need to look into that 👍.

To my knowledge, the prefix portion of a path is only relevant for data files, and perhaps the `file://` prefix is therefore not supported across all flags/config options. Have you tried using the `file:///c:/` prefix for your data files, and just the `c:/` prefix for your certificate? (if it is the case that the `file://` pattern isn't supported across all of OPA, that might also be something we should look into 😄).

From the sounds of it, maybe the issue is how non-data files, such as certs, are loaded on Windows. I haven't looked into the hot path for cert loading, though; maybe it all pipes into here(?).

Reply:
If I try to load the certificate without the `file:///` then I am back to where I started, 
My certificate path would be stripped from the prefix and I could only load certificates being in folders in the same drive as the OPA executable.
E.g. If my OPA executable is in the C:/ drive and I would like to load the certificate from `F:/cert/certificate.pem` then my certificate path would become` /cert/certificate.pem` and as long as this path is not present on the` C:/ `drive then I would get an error

Reply:
So, to sum things up:

1. the `file://` format isn't supported for certificates,
2. but certificate loading still utilizes `SplitPrefix`; stripping the drive letter as a prefix.
3. when the drive letter is stripped, Windows will default to `c:`(?)

Since data loading should be the only feature that cares about the prefix component, it sounds like the solution would be to make all other file reading - such as certs - not use `SplitPrefix`. (Additionally, the `file://` pattern should probably be supported across the board for consistency. But that's extra credits).

Inquiry for experimentation (🤔): with the assumption that the prefix is thrown away for cert loading, would it work to add a redundant prefix? E.g. `x:c:/a/b/c`, where `x` would be dropped and `c` remain in the path as drive letter. This wouldn't be a proper solution, of course, but maybe a means to unblock you if this is a production issue.

Reply:
For point 3 of your summary, the observed behaviour is that Windows defaults the drive letter as being the same one as where the process is being executed (if we run opa in `F:`, then is defaulted to `F:`, same for other partitions)

Using fake prefix(`x:c:/a/b/c`) does not do the trick, unfortunately.

After further investigation:

Even with the loading of data files, the `SplitPrefix` function is not always called, it's only used for the `PathWatcher`. When running the command with the `-w` flag then the `PathWatcher` is created for the data files, so if we run it without the flag, the `SplitPrefix` function is not called.

However, there is another function that seems to do the prefix stripping for the data files https://github.com/open-policy-agent/opa/blob/4b8787976755a2cc2147f62af97ce8ae2f6d65b5/internal/file/url/url.go#L18

This is being used right before loading the bundle.tar.gz

https://github.com/open-policy-agent/opa/blob/4b8787976755a2cc2147f62af97ce8ae2f6d65b5/loader/loader.go#L244

I think this is preventing the os error error: `open file:///C:/Temp/certs/certificate.pem: The filename, directory name, or volume label syntax is incorrect`

However this function is not called when loading the cert. My suggestion would be to add the same function call right before loading the certs.